### DISCUSSION TITLE
офф топ

### DISCUSSION THREAD

**Discussion by Kor-Angar** [dosinabox/g1_steam_fix#5] [General]
подскажи пожалуйста ты выложила "перевод" в воркшопе на стим. там gothic.dat снежков или ты их скрипты компилировала? 

есть декомпилированные скрипты снежков?

**Comment by dosinabox** [comment]
Чистый gothic.dat снежков, с диска, без модификаций.
Декомпилированных скриптов у меня нет.

**Reply by Kor-Angar** [reply]
> Чистый gothic.dat снежков, с диска, без модификаций.

значит надо будет сделать к ними мини мод на ин-экстремо :-D

**Comment by Kor-Angar** [comment]
а в готике 1 `Wld_RemoveNpc` работает как то иначе?

**Reply by dosinabox** [reply]
не знаю

**Reply by Kor-Angar** [reply]
попробовал. вроде работает

**Comment by Kor-Angar** [comment]
> Такой мод будет полезен и для английской версии

а там тоже почикали? :-) 

***

InExtremo for Russian Snowball

https://steamcommunity.com/sharedfiles/filedetails/?id=2811395389

**Reply by dosinabox** [reply]
> а там тоже почикали? :-)

Да.

Так что лучше назови мод **In Extremo Restoration**, убери требование русификации и напиши понятное описание на английском и русском.

Покажи код.

**Reply by Kor-Angar** [reply]
```D
// **************
// Kapitelwechsel
// **************

FUNC VOID B_Kapitelwechsel (VAR INT neues_Kapitel)
{
	B_Kapitelwechsel_old (neues_Kapitel);
	
	//***************************************************************************
	//	KAPITEL 2
	//***************************************************************************
	if	(Kapitel == 2)
	{
		//-------- InExtremo vorbereiten --------
		B_InsertInExtremo();
		
	};
	
	//***************************************************************************
	//	KAPITEL 3
	//***************************************************************************
	if (Kapitel == 3)
	{
		//-------- InExtremo wegrдumen --------
		B_InExtremoStopMusic();	
		B_KillInExtremo();
				
		// 	B_ExchangeRoutine			(Charlotte,					"off"); 
		// 	TA_Stay	(08,00,20,00,"NOWHERE");
		// 	смена расписания несрабатывает. возможно в мире нет "NOWHERE" ?
		
		Wld_RemoveNpc			(InExtremo_DrPymonte);	
		Wld_RemoveNpc			(InExtremo_TheFlail);	
		Wld_RemoveNpc			(InExtremo_ThomasTheForger);	
		Wld_RemoveNpc			(InExtremo_Unicorn); 	
		Wld_RemoveNpc			(InExtremo_YellowPfeiffer);	
		Wld_RemoveNpc			(InExtremo_Lutter);	
		Wld_RemoveNpc			(InExtremo_Flex);		
		Wld_RemoveNpc			(IE_397_Announcer); 
		Wld_RemoveNpc			(Charlotte); 

		Wld_RemoveNpc			(IEFan1); 
		Wld_RemoveNpc			(IEFan2); 
		Wld_RemoveNpc			(IEFan3); 
		Wld_RemoveNpc			(IEFan4); 
	};
};
```


```D
FUNC int  DIA_Grim_INEXTREMO_Condition()
{
	if	(Kapitel == 2) 
	{
		return TRUE;
	};
};


FUNC int  DIA_Grim_INEXTREMOAWAY_Condition()
{
	if	Npc_KnowsInfo(hero, DIA_Grim_INEXTREMO)
	&&	(Kapitel == 3) 
	{
		return TRUE;
	};
};
```


>  Так что лучше назови мод In Extremo Restoration, убери требование русификации и напиши понятное описание на английском и русском.

надо сначала проверить на английском. а то вдруг там вырезали под корень ;-)

**Reply by dosinabox** [reply]
> возможно в мире нет "NOWHERE" ?

Нету, в этом и задумка. Если нужно убрать персонажа, то его можно переместить на несуществующую точку.
Не сработает, если попытаться сделать это с персонажем в области видимости (возможно для проверки ты менял главу через помощника прямо перед сценой). В Г1 персонажи не телепортируются на точки для расписаний (если правильно помню), в отличие от Г2.

**Reply by Kor-Angar** [reply]
Наверняка автор "задумки" в армии не служил :-)

**Reply by Kor-Angar** [reply]
> надо сначала проверить на английском. а то вдруг там вырезали под корень ;-)

вроде в английской версии тоже работает

**Reply by Kor-Angar** [reply]
обновил превью и добавил ролик с английской версией

**Reply by dosinabox** [reply]
Что произойдет, если запустить мод на немецкой версии или в Руссобит? Появятся два концерта?

**Reply by Kor-Angar** [reply]
я не пробовал. скорее всего произойдёт два раза вызов функции `B_InsertInExtremo`. можно сделать конечно доп проверку на то есть ли в мире NPC и если их нет то вызывать B_InsertInExtremo.

**Reply by dosinabox** [reply]
Покажи DIA_Grim_INEXTREMO_Condition() и DIA_Grim_INEXTREMOAWAY_Condition() из чистых скриптов Snowball.

**Reply by Kor-Angar** [reply]
у меня скриптов Snowball нет. я брал из MDK.

**Reply by Kor-Angar** [reply]
диалоги с Grim ни на что не влияют. с ним можно вообще не разговаривать.

**Reply by dosinabox** [reply]
Просто интересно, каким образом это вырезано.

**Reply by dosinabox** [reply]
Понятно)
```cpp
func int DIA_Grim_INEXTREMO_Condition()
{
	return FALSE;
};
```

**Reply by Kor-Angar** [reply]
:-)

**Reply by Kor-Angar** [reply]
добавить ?

```D
if  Kapitel == 2
&& !Hlp_IsValidNpc (Charlotte)
{
	B_InsertInExtremo();
};

```

**Reply by dosinabox** [reply]
Моя реализация. Исключается вероятность двойного вызова функций и удалено всё лишнее.
[Patch_InExtremo.zip](https://github.com/dosinabox/g1_steam_fix/files/8779108/Patch_InExtremo.zip)

**Reply by Kor-Angar** [reply]
не вижу разницы. так же будет двойной вызов.

**Reply by dosinabox** [reply]
Старые функции отключаются. Вызываются новые с аналогичным содержимым.
Если у юзера **есть** B_InsertInExtremo в B_Kapitelwechsel, то ничего не произойдет. Затем вызывается Patch_InExtremo_Insert.
Если у юзера **нет** B_InsertInExtremo в B_Kapitelwechsel, то ничего не произойдет. Затем вызывается Patch_InExtremo_Insert.
В обоих случаях работает только новая функция и музыканты вставляются только один раз.

**Reply by Kor-Angar** [reply]
понятно. сразу не обратил внимание на пустые функции. 

добавил проверку в стили пираней 
```D
var int InsertInExtremo;

func void B_InsertInExtremo()
{
	if InsertInExtremo ==0
	{
		B_InsertInExtremo_old ();
		InsertInExtremo = 1;
	};
};
```

***

> Моя реализация. Исключается вероятность двойного вызова функций и удалено всё лишнее.

а это для какой версии ты делала?

**Reply by dosinabox** [reply]
> а это для какой версии ты делала?

Подойдет для всех. В английской и снежковой концерт активируется, а в остальных не сломается.

**Reply by Kor-Angar** [reply]
это я понял. а делала ты это для какой версии? и почему не выложила? :-)

**Reply by dosinabox** [reply]
Не понимаю вопрос про версию) 
Ты и выложи обновлением. Это твой код, я его причесала немного и исправила возможную двойную вставку.

**Reply by dosinabox** [reply]
> var int InsertInExtremo;

Если удалить мод, то эта переменная исчезнет из сохранения, а при повторной установке мода будет снова равняться 0.

**Reply by Kor-Angar** [reply]
теперь понятно. я думал это твой старый код из прошлых лет :-)

**Reply by Kor-Angar** [reply]
> Это твой код, я его причесала немного и исправила возможную двойную вставку.

нельзя всё лишнее удалять. ты оставляешь мало следов, для следующих поколений. лишаешь историков будущего **возможности** понять наш мир и быт, ход наших мыслей, наши надежды и чаяния. человек должен отличать от робота ;-)

**Reply by Kor-Angar** [reply]
вечером выложу.

**Reply by dosinabox** [reply]
Если хочешь оставить что-то для потомков, то пиши развернутые комментарии на английском. Всё остальное только затрудняет чтение кода и восприятие задумки, особенно непонятные закомментированные куски кода (ещё и с немецким текстом!). А в данном случае лишнее это ещё и функция DIA_Grim_INEXTREMOAWAY_Condition() аналогичная оригиналу, в моде она не нужна.

[Плохой комментарий хуже отсутствия комментария.](https://habr.com/ru/company/vk/blog/568610/)

**Reply by Kor-Angar** [reply]
херня. потомки на то и потомки что бы не быть современниками . писать нужно на том языке на каком мыслишь. если мыслишь по русски, то комментарий на английском  не сможет отобразить полёт мысли и исследователям будет не интересны. или не так интересны. в этом легко убедиться если послушать тех кто занимается историй ;-)  например клим жуков, очень интересно излагает про аспекты социальной инженерии в феодализме.

> затрудняет чтение кода и восприятие задумки, особенно непонятные закомментированные куски кода (ещё и с немецким текстом!).

о.. в этом и вся суть. ты даёшь первоисточник. первоисточник намного важнее чем чья бы ни была его интерпретация. дело не в том что он немецкий, дело в том что оригинальный. и вот когда на оригинальном тексте что то написано это передаёт мысль. английский перевод это уже интерпретация мысли ;-) которое уже само по себе несёт риск ошибки.

> лишнее это ещё и функция DIA_Grim_INEXTREMOAWAY_Condition()

ты этого знать не можешь. ты можешь предполагать это исходя из своего опыта.  и я могу предполагать. но что бы знать надо  проверить  код всех языковых версий. делать это трудозатратно. исходя из этого я её и оставил. а вдруг в какой то версии она закомментирована или всегда возвращает ноль? 
мой заф кафедры потратил бы час на то что бы вдолбить всему потоку что это единственно правильное решения, поясняя чем может закончится пренебрежение риском. вот в компьютерах чем такое заканчивается? ничем. патчем. а на производстве палец может отрезать или руку или глаз повредится или дыхательные пути. сколько здоровья сгубило лихость удалая. а пилоты что творят? мне вот из минска нравилось летать, там технология строго соблюдается  и обливают самолёт именно перед взлётом. а в шарики обливаю до начало движения. а по гостам поливать надо непосредственно перед взлётом когда двигатели основные уже запущены. к чему это приведёт? к тому что однажды закрылки заклинить и будет трагедия. это если кратко излагать.


> [Плохой комментарий хуже отсутствия комментария.]

этот вводный тезис в тексте не доказывается. более того он нивелируется в следующем же предложение. любой человек находящийся в тупике скажет, что наличие любого комментария лучше чем его отсутствия. любой человек. роботу любой комментарий не нужен. роботу лучше отсутствие, чем присутствие того что он обработать не может. этим человек и отличается от робота ;-)

**Reply by Kor-Angar** [reply]
> Моя реализация. Исключается вероятность двойного вызова функций и удалено всё лишнее.
> [Patch_InExtremo.zip](https://github.com/dosinabox/g1_steam_fix/files/8779108/Patch_InExtremo.zip)

выложил твою версию.

**Reply by Kor-Angar** [reply]
в твоей версии в англ версии в  3 главе не  пропадают НПС.

**Reply by Kor-Angar** [reply]
в версии от снежков тоже остаются в 3 главе.

**Reply by dosinabox** [reply]
А если менять главу в Болотном лагере?
А если поспать после смены главы в Старом лагере?

**Reply by dosinabox** [reply]
И попробуй ещё так:
```cpp
func void Patch_InExtremo_Remove()
{
	B_ExchangeRoutine(InExtremo_DrPymonte,"off");
	Wld_RemoveNpc(InExtremo_DrPymonte);
	B_ExchangeRoutine(InExtremo_TheFlail,"off");
	Wld_RemoveNpc(InExtremo_TheFlail);
	B_ExchangeRoutine(InExtremo_ThomasTheForger,"off");
	Wld_RemoveNpc(InExtremo_ThomasTheForger);
	B_ExchangeRoutine(InExtremo_Unicorn,"off");
	Wld_RemoveNpc(InExtremo_Unicorn);
	B_ExchangeRoutine(InExtremo_YellowPfeiffer,"off");
	Wld_RemoveNpc(InExtremo_YellowPfeiffer);
	B_ExchangeRoutine(InExtremo_Lutter,"off");
	Wld_RemoveNpc(InExtremo_Lutter);
	B_ExchangeRoutine(InExtremo_Flex,"off");
	Wld_RemoveNpc(InExtremo_Flex);
	B_ExchangeRoutine(IE_397_Announcer,"off");
	Wld_RemoveNpc(IE_397_Announcer);
	B_ExchangeRoutine(Charlotte,"off");
	Wld_RemoveNpc(Charlotte);
	B_ExchangeRoutine(IEFan1,"off");
	Wld_RemoveNpc(IEFan1);
	B_ExchangeRoutine(IEFan2,"off");
	Wld_RemoveNpc(IEFan2);
	B_ExchangeRoutine(IEFan3,"off");
	Wld_RemoveNpc(IEFan3);
	B_ExchangeRoutine(IEFan4,"off");
	Wld_RemoveNpc(IEFan4);
	B_ExchangeRoutine(Vlk_580_Grim,"start");
};
```

**Reply by Kor-Angar** [reply]
> И попробуй ещё так:

так работает.

**Reply by Kor-Angar** [reply]
![20220527014001_1](https://user-images.githubusercontent.com/49514764/170591161-228697af-b0ca-4cc2-a32c-ea8cd8e95ffe.jpg)

**Reply by dosinabox** [reply]
Это где, это как?

**Reply by Kor-Angar** [reply]
теперь понятно почему Пираньи так любят подобные проверки

```D
var int InsertInExtremo;

func void B_InsertInExtremo()
{
	if InsertInExtremo ==0
	{
		B_InsertInExtremo_old ();
		InsertInExtremo = 1;
	};
};
```
![20220527014001_1](https://user-images.githubusercontent.com/49514764/170591312-6ac2ccc3-20c1-40af-b35a-49f54be4364d.jpg)

немцы конечно молодцы. дисциплина. дисциплина. дисциплина.

**Reply by Kor-Angar** [reply]
> Это где, это как?

стори хелпер. движение по сюжету.

**Reply by dosinabox** [reply]
Думаю, при движении по сюжету так же будет и в оригинале.

**Reply by Kor-Angar** [reply]
не будет. там же проверка стоит что бы прогружалось один раз.

**Reply by Kor-Angar** [reply]
ан нет. в готики 1 таких проверок нет. они появились во второй части

**Reply by Kor-Angar** [reply]
> Думаю, при движении по сюжету так же будет и в оригинале.

будут

**Reply by Kor-Angar** [reply]
[Patch_InExtremo_SE.zip](https://github.com/dosinabox/g1_steam_fix/files/8782970/Patch_InExtremo_SE.zip)

**Reply by dosinabox** [reply]
Лучше переименовать B_RemoveNpc во что-то другое, например в B_RemoveInExtremoMember. Во-первых, эта функция работает не с любыми NPC, а только с **некоторыми**, у которых есть расписание "off". Во-вторых, у игрока может быть установлен какой-нибудь мод, патч или плагин, где тоже будет функция B_RemoveNpc. Это название слишком общее.
То же с B_InsertNpc. Если функцию используешь только ты и только в определенных местах, то постарайся дать ей более отличительное название.

Зачем нужен комментарий, просто повторяющий название функции? Емкое и понятное название вместе с аргументами - уже достаточно информативно само по себе.
```cpp
// --------------------------------------- Insert Npc -------------------------------------
func void B_InsertNpc (var int npcInstance,var string  spawnpoint)
```
```cpp
// --------------------------------------- Remove Npc -------------------------------------
func void B_RemoveNpc (var int npcInstance)
```
А здесь комментарии не соответствует содержимому. info в скриптах это func void с содержимым диалога (реплики, субтитры, действия), а в моде меняются условия func int + _Condition (логика доступности). И DIA_Grim_INEXTREMOAWAY_Condition отличается от оригинала (условия расширены) - почему бы не написать в комментарии, что именно ты поменял и почему?
```cpp
//---------------------------------------------------------------------
//	Info INEXTREMO
//---------------------------------------------------------------------
func int DIA_Grim_INEXTREMO_Condition()
```
```cpp
//---------------------------------------------------------------------
//	Info INEXTREMOAWAY
//---------------------------------------------------------------------
func int  DIA_Grim_INEXTREMOAWAY_Condition()
```

B_INSERTINEXTREMO.D, DIA_VLK_580_GRIM.D - [хорошим тоном считается](https://semakin.dev/2020/05/no_newline_at_end_of_file/) оставлять в конце текстового файла пустую строку.

**Reply by Kor-Angar** [reply]
> Если функцию используешь только ты и только в определенных местах, то постарайся дать ей более отличительное название.

это факт :-) у меня был префикс 2022, но он не везде компилируется. видимо в каких то местах кода цифры воспринимаются как отсылка к id npc,

> Зачем нужен комментарий, просто повторяющий название функции? Емкое и понятное название вместе с аргументами - уже достаточно информативно само по себе.

там не одна функция , а три. комментарий визуально показывает начало блока.

> А здесь комментарии не соответствует содержимому. info в скриптах это func void с содержимым диалога (реплики, субтитры, действия), а в моде меняются условия func int + _Condition (логика доступности).

это "разметка" кода из MDK. в данном случае `info ` не имеет отношение к `func void `, а является существительным **info** 

> Лучше переименовать

переименовал


> оставлять в конце текстового файла пустую строку.

добавил

> И DIA_Grim_INEXTREMOAWAY_Condition отличается от оригинала (условия расширены) - почему бы не написать в комментарии, что именно ты поменял и почему?

по русски? :-)  добавлено еще одно условие: на тот случай если герой не поговорил с Grim во второй главе, но при этом "запустил" концерт ИнЭкстримо. В таком случае герой сможет спросить у Grim про ИнЭкстримо в 3 главе.

```D
		if	Npc_KnowsInfo(hero, DIA_Grim_INEXTREMO)			// MDK
		||	Npc_KnowsInfo(hero, IE_397_Announcer_ANNOUNCE)	// patch
		{
				return TRUE;
		};
```


***

когда ты пишешь **оригинал** понять возможно только из контектса. иногда ты имеешь ввиду MDK, иногда Акелла, иногда английские скрипты. в моём понимание оригинал это MDK.

**Reply by Kor-Angar** [reply]
[Patch_InExtremo_SE.zip](https://github.com/dosinabox/g1_steam_fix/files/8786302/Patch_InExtremo_SE.zip)

**Reply by dosinabox** [reply]
> у меня был префикс 2022

Названия переменных и функций не могут начинаться с цифр и содержать пробелы. Наверное, во всех языках программирования.

**Comment by dosinabox** [comment]
AutoScreenRes.dll для Г1 тоже было бы неплохо загрузить в мастерскую

**Reply by Kor-Angar** [reply]
а файл тот же?

**Reply by dosinabox** [reply]
да

**Reply by Kor-Angar** [reply]
добавил

**Reply by Kor-Angar** [reply]
https://steamcommunity.com/sharedfiles/filedetails/?id=2812457240

**Comment by Kor-Angar** [comment]
_- Основная причина случившегося – нарушение правил обработки воздушных судов противообледенительной жидкостью. Чтобы обработать ею такой громадный самолёт, нужно 230 литров этой смеси. А там было израсходовано только 90!_

_Но и это ещё не всё. При обработке самолета жидкостью почему-то было принято решение облить только крылья, стабилизатор, но не трогать фюзеляж. В результате получилось так: пока самолет двигался по рулежке, кабина нагрелась, на ней начал таять снег – в тот день был очень сильный снегопад – струйки воды стекали вниз под фюзеляж. А там установлены трубки приема давления воздуха. Они принимают воздушный поток, выдавая показатели скорости на приборе пилота. Затем эти же данные получает автопилот, который в зависимости от них регулирует действия самолёта._

https://www.mk.ru/politics/2022/01/25/zasluzhennyy-pilot-nazval-osnovnye-problemy-rossiyskoy-aviacii-diletantstvo-i-korrupciya.html

**Comment by Kor-Angar** [comment]
```D
/*
// *******************************************************************
        AI_Output(self,hero,"Info_Diego_Gamestart_11_01"); //Warum haben sie dich hier reingeworfen?
        
        Info_ClearChoices(Info_Diego_Gamestart );
        Info_AddChoice   (Info_Diego_Gamestart, "Gewitzter Ьberfall            (+3 Geschicklichkeit)"   ,Info_Diego_Gamestart_Choice_Dex);
        Info_AddChoice   (Info_Diego_Gamestart, "Schlдgerei mit Folgen         (+3 Stдrke)"                             ,Info_Diego_Gamestart_Choice_Str);
        Info_AddChoice   (Info_Diego_Gamestart, "Wahre Worte zum falschen Mann (+3 Mana)"                               ,Info_Diego_Gamestart_Choice_Man);
        Info_AddChoice   (Info_Diego_Gamestart, "Unschuldig                    (+1 auf alle Attribute)" ,Info_Diego_Gamestart_Choice_All);
};
// -------------------------------------------- CHOICES ----------------------------------------------------
FUNC VOID Info_Diego_Gamestart_Choice_Dex()
{
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_Dex_15_00"); //Ich habe ne Ladung Erz verschwinden lassen.
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_Dex_15_01"); //Dummerweise hat der Typ bei dem ich sie versteckt habe, lieber die Belohnung fьr mich kassiert, anstatt mit mir das Erz zu teilen.
        PrintScreen     ("Geschicklichkeit +3", -1,-1,"FONT_OLD_20_WHITE.TGA",3);
        hero.attribute[ATR_DEXTERITY] = hero.attribute[ATR_DEXTERITY]+3;
        Info_ClearChoices(Info_Diego_Gamestart );
};
FUNC VOID Info_Diego_Gamestart_Choice_Str()
{
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_Str_15_00"); //Eine Herausforderung konnte ich noch nie abschlagen. Auch nicht von des Kцnigs Gardisten. dummerweise hat der Kцnig viele Gardisten.
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_Str_15_01"); //Zu viele.       
        PrintScreen     ("Stдrke +3", -1,-1,"FONT_OLD_20_WHITE.TGA",3);
        hero.attribute[ATR_STRENGTH] = hero.attribute[ATR_STRENGTH]+3;
        Info_ClearChoices(Info_Diego_Gamestart );
};
FUNC VOID Info_Diego_Gamestart_Choice_Man()
{
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_Man_15_00"); //Ich saЯ in der Kneipe und habe meine Meinung ьber Kцnig, Krieg und Steuern gesagt. 
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_Man_15_01"); //Ich hab' mich nur zu spдt daran erinnert, dass der, der die Wahrheit spricht, ein schnelles Pferd braucht.
        PrintScreen     ("Mana +3", -1,-1,"FONT_OLD_20_WHITE.TGA",3);
        hero.attribute[ATR_MANA]                = hero.attribute[ATR_MANA]+3;
        hero.attribute[ATR_MANA_MAX]    = hero.attribute[ATR_MANA_MAX]+3;
        Info_ClearChoices(Info_Diego_Gamestart );
};
FUNC VOID Info_Diego_Gamestart_Choice_All()
{
        AI_Output(hero,self,"Info_Diego_Gamestart_Choice_All_15_00"); //Ich bin unschuldig!
        AI_Output(self,hero,"Info_Diego_Gamestart_Choice_All_11_01"); //Du solltest eins wissen: Hier drin ist niemand unschuldig. 
        AI_Output(self,hero,"Info_Diego_Gamestart_Choice_All_11_02"); //Hier findest du Mдnner die so ziemlich jedes mцgliche Verbrechen begangen haben. Und einen unschuldigen nehmen die nur zu gerne auseinander. 
        AI_Output(self,hero,"Info_Diego_Gamestart_Choice_All_11_03"); //Ich gebe dir nen Tipp, und der ist ganz umsonst. 
        AI_Output(self,hero,"Info_Diego_Gamestart_Choice_All_11_04"); //Wenn dich jemand fragt, warum du hier bist, sag ihm, du hast einen Typen gekillt, der dich gefragt hat, warum du hier bist. 
        PrintScreen     ("Stдrke, Geschick und Mana +1", -1,-1,"FONT_OLD_20_WHITE.TGA",3);
        hero.attribute[ATR_DEXTERITY]   = hero.attribute[ATR_DEXTERITY]+1;
        hero.attribute[ATR_STRENGTH]    = hero.attribute[ATR_STRENGTH]  +1;
        hero.attribute[ATR_MANA]                = hero.attribute[ATR_MANA]              +1;
        hero.attribute[ATR_MANA_MAX]    = hero.attribute[ATR_MANA_MAX]  +1;
        Info_ClearChoices(Info_Diego_Gamestart );
*/
```

**Comment by Kor-Angar** [comment]
```D
	Scatty: 	Scatty_ChargeKirgo == LOG_SUCCESS 	+2
				Scatty_ChargeKirgo == LOG_FAILED 	-1
				Scatty_ChargeKharim == LOG_SUCCESS 	+5
				Scatty_ChargeKharim == LOG_FAILED 	+2
			
	Whistler: 	Whistler_BuyMySword = LOG_SUCCESS;	+3
				Whistler_BuyMySword = LOG_FAILED;	+2
				
	Sly: 		Sly_LostNek = LOG_SUCCESS;			+4
	
	Fingers: 	Fingers_Learnt = LOG_SUCCESS; 		+3
	
	Dexter: 	Dexter_GetKalomsRecipe = LOG_SUCCESS; +5
	
	Thorus: 	Thorus_MordragKo = LOG_SUCCESS;		+5
				Thorus_MordragKo = LOG_FAILED;		-5
```
судя по минусам, пираньи закладывали вариант при котором игрок может завалить вступление в СЛ :-) видимо запасным аэродром является вступление в НЛ.

**Reply by Kor-Angar** [reply]
если -5 и -1, то остаётся +2+4+3+5
-6+6+8
итого +8

**Reply by Kor-Angar** [reply]
ну да. на вступление надо 15. при 8 пойнтах вообще вариант.

**Reply by Kor-Angar** [reply]
если просто завалить задание торуса, то +2+5+3+4+3+5-5, то есть
+2+5+3+4+3
+10+4+3
шанс не большой. гипотетически возможно если сильно постараться. но можно и завалить :-)

**Comment by Kor-Angar** [comment]
```D

// Alte Version mit AI-Outputs
/*	    var int talktime;
    	talktime = Hlp_Random (200);
    	if 		(talktime <   5)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text01");//Und was glaubst Du, wie es war? Es war ein Trauerspiel. Erst platzt dieser Kerl rein, die Waffe gezogen. Stinksauer.
    	} 
    	else if (talktime <  10)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text02");//Und just in dem Moment stolpert er ьber eine Wurzel! Ich hab mich plattgelacht. Sein Gesicht voller Kratzer. Es war herrlich.
    	}
    	else if (talktime <  15)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text03");//Auf einmal greift er in den Topf mit dem Reis und sagt: Ist das Bier wirklich hier drin? Und verbrennt sich die Finger, daЯ er schreit!
    	}
    	else if (talktime < 20)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text04");//Ich weiЯ auch nicht, ob das stimmt. Es wird immer so viel geschwдtzt, aber keiner weiЯ was genaues. Und uns sagt eh keiner was.
    	}
    	else if (talktime < 25)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text05");//Warum sollte er das tun? Er wьrde es sich ganz bestimmt mit ihnen verscherzen. Das wagt er nicht. Das ist ein Feigling.
    	}
    	else if (talktime < 30)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text06");//Ach was! Du glaubst auch alles, was man Dir erzдhlt. Fьr wie dumm hдlst Du mich eigentlich? Denk doch mal nach. Dafьr hдtte die Zeit nicht gereicht.
    	}
    	else if (talktime < 35)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text07");//Ist mir eigentlich auch egal. Wen interessierts? Wenn Du Deine Nase nicht reinsteckst, gibt es auch keinen Дrger. Und das ist besser so.
    	}
    	else if (talktime < 40)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text08");//Und dann springt er auf einmal in meinen Weg und sagt: Der Gobbo war's! Da hab ich ihm gesagt, er soll sich verpissen. Gobbo. Dem geb ich Gobbo.
    	}
    	else if (talktime < 45)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text09");//Ich hab ihm kein Wort geglaubt. Ein Shadowbeast? Dieser Schwдchling? Im Leben nicht! Vermutlich nur eine mittelschwere Ratte.
    	}
    	else if (talktime < 50)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text10");//Da hab ich ihn gefragt, woher er das wisse. Und weiЯt Du was er gesagt hat? Du errдtst es nicht... Aus einem Brief! Der kann doch weder lesen, noch kennt er jemanden, der schreiben kann!
    	}
    	else if (talktime < 55)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text11");//Verdammt noch mal nein. Ich hab's Dir schon tausendmal erklдrt. Was soll die Fragerei? Langsam muЯt Du's doch mal kapieren. Ich war zu der Zeit in der Mine.
    	}
    	else if (talktime < 60)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text12");//Da gab's dann nur noch ein Problem bei der Sache. Der Kerl stand bei ihr im Schrank und hat alles mit angehцrt! Der Rest des Abends war gelaufen.
    	}
    	else if (talktime < 65)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text13");//Ich glaube, das Wetter wird wieder besser. Jetzt wo wir bald wieder in die Mine mьssen. Hervorragend. Damit gehцrt das Bier mir.
    	}
    	else if (talktime < 70)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text14");//Vor Lachen hau ich mir auf's Knie! Du und die Arena, sag ich. Du weiЯt ja nicht mal, wo beim Schwert der Griff sitzt.
    	}
    	else if (talktime < 75)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text15");//Kannst Du Dir vorstellen, was da alles lagert? Kein Wunder, daЯ er niemanden mitnimmt. Das ist alles gut versteckt, sicher.
    	}
    	else if (talktime < 80)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text16");//Ich hab nix gesehen, aber ich hab so ein Gefьhl. Das war er. Es war zwar stockdunkel, aber ich bin sicher. Er war's. Kein ScheiЯ. Ich weiЯ auch nicht, wie er das angestellt hat.
    	}
    	else if (talktime < 85)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text17");//Da sagt er mir, das sei gar keine Gefahr. Ich solle einfach vorgehen, er wьrde mir den Rьcken freihalten. Klar! Ich sag ihm, vergiЯ es. Du gehst vor.
    	}
    	else if (talktime < 90)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text18");//Plцtzlich fдllt ihm die Spitzhacke aus der Hand und er lдuft lila an. Ich hau ihm auf den Rьcken und weiЯt Du, was er verschluckt hat? Erz! Wollt er wohl rausschmuggeln.
    	}
    	else if (talktime < 95)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text19");//Nein, er doch nicht, sagt er entrьstet! Er kennt sich mit Pilzen aus, sagt er. Ich sag danke und verpiЯ mich. Und was war? Vier Tage lag er da, mehr tot als lebendig.
    	}
    	else if (talktime < 100)
    	{
    		AI_Output (self,NULL,"ZS_SmallTalk_Text20");//Ich frag ihn wo er hinwill. Er sagt, das geht mich einen feuchten ScheiЯdreck an. Der war auf dem Weg zum Versteck. Wetten? Dem kannste nicht trauen.
    	};
*/
```
